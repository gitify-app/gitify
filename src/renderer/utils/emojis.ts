import twemoji from '@discordapp/twemoji';
import { Constants } from './constants';
import { Errors } from './errors';

const ALL_EMOJIS = [
  ...Constants.ALL_READ_EMOJIS,
  ...Errors.BAD_CREDENTIALS.emojis,
  ...Errors.MISSING_SCOPES.emojis,
  ...Errors.NETWORK.emojis,
  ...Errors.RATE_LIMITED.emojis,
  ...Errors.UNKNOWN.emojis,
];

export const EMOJI_CODE_POINTS = ALL_EMOJIS.map((emoji) =>
  twemoji.convert.toCodePoint(emoji),
);
