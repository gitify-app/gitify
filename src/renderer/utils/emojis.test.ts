import { ALL_EMOJI_SVG_FILENAMES } from './emojis';

describe('renderer/utils/emojis.ts', () => {
  it('emoji svg filenames', () => {
    expect(ALL_EMOJI_SVG_FILENAMES).toHaveLength(20);
    expect(ALL_EMOJI_SVG_FILENAMES).toMatchSnapshot();
  });
});
