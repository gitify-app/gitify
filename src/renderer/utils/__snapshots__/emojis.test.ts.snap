// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer/utils/emojis.ts emoji svg filenames 1`] = `
[
  "1f389.svg",
  "1f38a.svg",
  "1f973.svg",
  "1f44f.svg",
  "1f64c.svg",
  "1f60e.svg",
  "1f3d6.svg",
  "1f680.svg",
  "2728.svg",
  "1f3c6.svg",
  "1f513.svg",
  "1f52d.svg",
  "1f6dc.svg",
  "1f62e-200d-1f4a8.svg",
  "1f914.svg",
  "1f972.svg",
  "1f633.svg",
  "1fae0.svg",
  "1f643.svg",
  "1f648.svg",
]
`;
